src/lib/geospatial.ts

```diff
@@ -0,0 +1,43 @@
+interface LatLng {
+  lat: number
+  lng: number
+}
+
+const EARTH_RADIUS_IN_METERS = 6371e3 // https://en.wikipedia.org/wiki/Earth_radius#Mean_radius
+
+/**
+ * Convert angular distance from degrees to radians
+ *
+ * @param {number} degrees - decimal degrees
+ * @returns {number} radians
+ */
+const toRadians = (degrees: number): number => degrees * (Math.PI / 180)
+
+/**
+ * Calculates the haversine (spherical) distance between two geographic points.
+ * Arguments are supplied as `LatLng` objects, with coordinates specified
+ * as `lat` and `lng` properties in decimal degrees.
+ *
+ * See:
+ * https://en.wikipedia.org/wiki/Haversine_formula
+ * https://www.movable-type.co.uk/scripts/latlong.html
+ *
+ * @param {LatLng} point1 - an object with `lat` and `lng` properties
+ * @param {LatLng} point2 - an object with `lat` and `lng` properties
```

Comment:
Rather than documenting these objects here, I’d document the properties of the `LatLng` interface.